internal/api/synchronizer.proto

```diff
@@ -17,87 +17,21 @@ package openmatch.internal;
 option go_package = "open-match.dev/open-match/internal/ipb";
 
 import "api/messages.proto";
-import "google/api/annotations.proto";
-import "protoc-gen-swagger/options/annotations.proto";
 
-option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
-  info: {
-    title: "Synchronizer"
-    version: "1.0"
-    contact: {
-      name: "Open Match"
-      url: "https://open-match.dev"
-      email: "open-match-discuss@googlegroups.com"
-    }
-    license: {
-      name: "Apache 2.0 License"
-      url: "https://github.com/googleforgames/open-match/blob/master/LICENSE"
-    }
-  }
-  external_docs: {
-    url: "https://open-match.dev/site/docs/"
-    description: "Open Match Documentation"
-  }
-  schemes: HTTP
-  schemes: HTTPS
-  consumes: "application/json"
-  produces: "application/json"
-  responses: {
-    key: "404"
-    value: {
-      description: "Returned when the resource does not exist."
-      schema: { json_schema: { type: STRING } }
-    }
-  }
-  // TODO Add annotations for security_defintiions.
-  // See
-  // https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/proto/examplepb/a_bit_of_everything.proto
-};
-
-message RegisterRequest {
+message SynchronizeRequest {
+  openmatch.Match proposal = 1;
 }
 
-message RegisterResponse {
-  // Identifier for this request valid for the current synchronization cycle.
-  string id = 1;
-}
-
-message EvaluateProposalsRequest {
-  // List of proposals to evaluate in the current synchronization cycle.
-  openmatch.Match match = 1;
-
-  // Identifier for this request issued during request registration.
-  string id = 2;
-}
-
-message EvaluateProposalsResponse {
-  // Results from evaluating proposals for this request.
-  openmatch.Match match = 1;
+message SynchronizeResponse {
+  bool start_mmfs = 1;
+  bool cancel_mmfs = 2;
+  openmatch.Match match = 3;
 }
```

Comment:
Can we make this even simpler. Consider SynchronizeResponse only returns a match stream. 
- The Synchrnoize() call can block till you are ready to run MMFs - so once synchronize returns, the backend can trigger the MMF
- Once the proposal collection interval is over and evaluation done, we close the stream - and the backend automatically cancels any pending mmf. 

That way, the interaction between backend and synchronizer just stays an input / output stream and no need for explicit start / cancel actions overloaded on the stream response?

Reply:
I guess then for the register, this is the best option (short of separating out into a separate blocking register call). I will discuss the cancel condition offline on whether it can be handled without explicitly sending any information to the backend.